### PR TITLE
Revert "Revert "sql/cockroach: make postgres inspect compatible with crdb""

### PR DESCRIPTION
**Pull request by hedwigz**
Reverts ariga/atlas#819

### CODE REVIEW

File: sql/postgres/driver.go [hunk 1]
```diff
@@ -36,6 +36,8 @@ type (
 		collate string
 		ctype   string
 		version string
+		// is cockroachdb
```
Comment (a8m) [inline sql/postgres/driver.go:39]:
```suggestion
```

File: sql/postgres/driver.go [hunk 2]
```diff
@@ -216,6 +219,7 @@ const (
 	TypePath    = "path"
 	TypePolygon = "polygon"
 	TypePoint   = "point"
+	TypeGeometry = "geometry" // cockroach geometry.
```
Comment (a8m) [inline sql/postgres/driver.go:223]: gofmt

File: sql/postgres/inspect.go [hunk 3]
```diff
@@ -344,6 +360,77 @@ func (i *inspect) indexes(ctx context.Context, s *schema.Schema) error {
...
 
+var reIndexType = regexp.MustCompile("(?i)USING (BTREE|GIN|GIST)")
+
+// indexType extracts index type information from index create statement. see: https://www.cockroachlabs.com/docs/stable/create-index.html
+func indexType(createStmt string) *IndexType {
+	t := reIndexType.FindStringSubmatch(createStmt)
+	return &IndexType{T: t[1]}
+}
+
+func (i *inspect) crdbAddIndexes(s *schema.Schema, rows *sql.Rows) error {
```
Comment (a8m) [inline sql/postgres/inspect.go:371]: Please add a test case for this scanning logic. It will be easier to understand this block when looking at the test.

File: sql/postgres/inspect.go [hunk 4]
```diff
@@ -344,6 +360,77 @@ func (i *inspect) indexes(ctx context.Context, s *schema.Schema) error {
...
+	// NOTE: cockroach index names aren't unique
```
Comment (a8m) [inline sql/postgres/inspect.go:372]:
```suggestion
	// Cockroach index names aren't unique.
```

File: sql/postgres/inspect.go [hunk 5]
```diff
@@ -972,7 +1096,7 @@ ORDER BY
 SELECT
 	rel.relname AS table_name,
 	t1.conname AS constraint_name,
-	pg_get_expr(t1.conbin, to_regclass(quote_ident(nsp.nspname) || '.' || quote_ident(rel.relname))::oid) as expression,
+	pg_get_expr(t1.conbin, nsp.oid) as expression,
```
Comment (a8m) [inline sql/postgres/inspect.go:1099]: Why did you change this code query if you don't use it?

File: sql/postgres/inspect.go [hunk 6]
```diff
@@ -885,19 +973,55 @@ SELECT
...
+	t.relname AS table_name,
+	i.relname AS index_name,
+	a.attname AS column_name,
+	idx.indisprimary AS primary,
+	idx.indisunique AS unique,
+	c.contype AS constraint_type,
+			 pgi.indexdef create_stmt,
+	pg_get_expr(idx.indpred, idx.indrelid) AS predicate,
+	pg_get_indexdef(idx.indexrelid, idx.ord, false) AS expression,
+	 pg_catalog.obj_description(i.oid, 'pg_class') AS comment
```
Comment (a8m) [inline sql/postgres/inspect.go:1002]:
```suggestion
	pg_get_expr(idx.indpred, idx.indrelid) AS predicate,
	pg_get_indexdef(idx.indexrelid, idx.ord, false) AS expression,
	pg_catalog.obj_description(i.oid, 'pg_class') AS comment
```

File: sql/postgres/inspect.go [hunk 7]
```diff
@@ -344,6 +360,77 @@ func (i *inspect) indexes(ctx context.Context, s *schema.Schema) error {
...
+			}
+			if sqlx.ValidString(comment) {
+				idx.Attrs = append(idx.Attrs, &schema.Comment{Text: comment.String})
+			}
+			if sqlx.ValidString(contype) {
+				idx.Attrs = append(idx.Attrs, &ConType{T: contype.String})
+			}
+			if sqlx.ValidString(pred) {
+				idx.Attrs = append(idx.Attrs, &IndexPredicate{P: pred.String})
+			}
```
[the remark this answers is not in the document]
Comment (hedwigz) [reply sql/postgres/inspect.go:406]: טקד

File: sql/postgres/inspect.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (hedwigz) [reply sql/postgres/inspect.go]: fixed, and added test

File: internal/integration/mysql_test.go [hunk 8]
```diff
@@ -387,6 +387,22 @@ func TestMySQL_ColumnBool(t *testing.T) {
 	})
 }
 
+func TestMySQL_ColumnCheck(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (a8m) [reply internal/integration/mysql_test.go:390]: Use myRun instead.

File: internal/integration/sqlite_test.go [hunk 9]
```diff
@@ -71,6 +71,22 @@ func TestSQLite_Ent(t *testing.T) {
 	})
 }
 
+func TestSQLite_ColumnCheck(t *testing.T) {
```
Comment (a8m) [inline internal/integration/sqlite_test.go:74]: Same

File: sql/postgres/inspect_test.go [hunk 10]
```diff
@@ -182,6 +185,61 @@ users           | idx6            | brin        | c1          | f       | t
...
 		},
+		{
+			name: "crdb table indexes",
+			crdb: true,
+			before: func(m mock) {
+				m.tableExists("public", "users", true)
+				m.ExpectQuery(queryColumns).
+					WithArgs("public", "users").
+					WillReturnRows(sqltest.Rows(`
+				table_name | column_name | data_type | is_nullable |              column_default               | character_maximum_length | numeric_precision | datetime_precision | numeric_scale | character_set_name | collation_name | udt_name | is_identity | identity_start | identity_increment |       identity_generation        | generation_expression | comment | typtype | oid
```
Comment (a8m) [inline sql/postgres/inspect_test.go:196]: Keep indentations like all other test cases.

File: sql/postgres/inspect.go [hunk 11]
```diff
@@ -344,6 +360,77 @@ func (i *inspect) indexes(ctx context.Context, s *schema.Schema) error {
...
+var reIndexType = regexp.MustCompile("(?i)USING (BTREE|GIN|GIST)")
+
+// indexType extracts index type information from index create statement. see: https://www.cockroachlabs.com/docs/stable/create-index.html
+func indexType(createStmt string) *IndexType {
+	t := reIndexType.FindStringSubmatch(createStmt)
+	return &IndexType{T: t[1]}
+}
+
+func (i *inspect) crdbAddIndexes(s *schema.Schema, rows *sql.Rows) error {
+	// Cockroach index names aren't unique.
```
Comment (a8m) [inline sql/postgres/inspect.go:372]:
```suggestion
	// Unlike Postgres, Cockroach may have duplicate index names.
```
